src/commands/deploy.ts

```diff
@@ -152,26 +160,90 @@ export const run: Run<DeployArgs> = async (ctx) => {
+          });
+
+          const updated = chalk.blue("updated");
+          const deleted = chalk.red("deleted");
+
+          const rows: { symbol: string; name: string; action: string; indent: number }[] = [];
+
+          deletedModels.forEach((model: string) => {
+            rows.push({ symbol: deletedSymbol, name: chalk.red(model), action: deleted, indent: 0 });
+          });
+
+          deletedModelFields.forEach(({ modelIdentifier, fields }) => {
+            rows.push({ symbol: updatedSymbol, name: chalk.blue(modelIdentifier), action: updated, indent: 0 });
+            fields.forEach((field) => {
+              rows.push({ symbol: deletedSymbol, name: chalk.red(field), action: deleted, indent: 2 });
+            });
+          });
+
+          const longestNameLength = rows.reduce((longest, row) => Math.max(longest, row.name.length), 0);
+          const longestIndent = rows.reduce((longest, row) => Math.max(longest, row.indent), 0);
+          const indentSize = 2;
+
+          println({
+            ensureEmptyLineAbove: true,
+            content: chalk.gray("These changes will be applied to production based on the app you're deploying."),
+          });
+          for (const row of rows) {
+            const indentation = " ".repeat(row.indent * indentSize);
+            const namePadding = " ".repeat(longestNameLength - row.name.length + 2);
+            const actionPadding = " ".repeat((longestIndent - row.indent) * indentSize);
+            println({
+              ensureEmptyLineAbove: false,
+              content: indentString(`${indentation}${row.symbol} ${row.name}${namePadding}${actionPadding}${row.action}`, 6),
+            });
+          }
+        }
 
         if (!publishStarted) {
           await confirm("Do you want to continue?");
-          subscription.resubscribe({ ...variables, force: true });
+          subscription.resubscribe({ ...variables, force: true, allowDeletedData: true });
         } else {
-          assert(ctx.args["--allow-problems"], "expected --allow-problems to be true");
+          const allowDataDelete = ctx.args["--allow-data-delete"];
+          const allowProblems = ctx.args["--allow-problems"];
+
+          if (!allowDataDelete && !allowProblems) {
+            throw new UnexpectedError("expected --allow-data-delete or --allow-problems to be true");
+          }
+
+          const messages: string[] = [];
+
+          if (allowProblems) {
+            messages.push(sprint`Deploying regardless of problems because "${chalk.gray("--allow-problems")}" was passed.`);
+          }
+
+          if (allowDataDelete) {
+            messages.push(sprint`Deploying regardless of deleted data because "${chalk.gray("--allow-data-delete")}" was passed.`);
+          }
+
           println({
             ensureEmptyLineAbove: true,
-            content: sprint`Deploying regardless of problems because {bold "--allow-problems"} was passed.`,
+            content: sprint`${messages.join("\n")}`,
```

Comment:
nit: might be easier to just call `println` within the if blocks, rather than building up the messages.

Reply:
🙈